### PR TITLE
imaplib.pyi: Add ssl_context parameter for Python >= 3.3

### PR DESCRIPTION
**Pull request by utkarsh2102**
Python 3.3 onward, `ssl_context` was added; documented [here](https://docs.python.org/3/library/imaplib.html#imaplib.IMAP4_SSL).

Fixes: #3289 (the second part of the issue!)

### CODE REVIEW

File: stdlib/2and3/imaplib.pyi [hunk 1]
```diff
@@ -89,7 +89,11 @@ class IMAP4:
 class IMAP4_SSL(IMAP4):
     keyfile: str = ...
     certfile: str = ...
-    def __init__(self, host: str = ..., port: int = ..., keyfile: Optional[str] = ..., certfile: Optional[str] = ...) -> None: ...
+    ssl_context: str = ...
```
Comment (srittau) [inline stdlib/2and3/imaplib.pyi]: This is a `ssl.SSLContext` object, according to the documentation.

File: stdlib/2and3/imaplib.pyi [hunk 2]
```diff
@@ -89,7 +89,11 @@ class IMAP4:
...
+    if sys.version_info >= (3, 3):
+        def __init__(self, host: str = ..., port: int = ..., keyfile: Optional[str] = ..., certfile: Optional[str] = ..., ssl_context: Optional[str] = ...) -> None: ...
```
Comment (srittau) [inline stdlib/2and3/imaplib.pyi]: Here too.

Comment (utkarsh2102) [reply stdlib/2and3/imaplib.pyi]: Yikes, fixed.

**Comment by utkarsh2102** [comment]
Hey @srittau,  
How do I go on solving: `Name 'ssl' is not defined`?

**Comment by srittau** [comment]
Just use `SSLContext`. No need to prefix it with `ssl`.

**Pull request merged**